lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2140,162 +2148,225 @@ static Value createLinalgPayloadCalculationForReduceOp(
         convertScalarToDtype(b, loc, payloadArgs[0], resultElementType);
     Value result = payloadArgs[1];
     return b.create<arith::AddFOp>(loc, self, result);
+  } else if (isa<AtenMaxOp>(op) && resultElementType.isa<mlir::FloatType>()) {
+    Value self =
+        convertScalarToDtype(b, loc, payloadArgs[0], resultElementType);
+    Value result = payloadArgs[1];
+    return b.create<arith::MaxFOp>(loc, self, result);
   }
   op->emitError("unimplemented lowering in "
                 "createLinalgPayloadCalculationForReduceOp");
   return nullptr;
 }
 
-namespace {
-// Aten argmax lowering represents the ArgMax op as an linalg.indexed_generic
-// op, producing two output buffers.
+// `createArgMaxKernel` returns two values:
+// The first output buffer contains the maximum value found. It is initialized
+// to the minimum representable value of the input element type. //
 //
-// The first output buffer contains the index of the found maximum value. It is
+// The second output buffer contains the index of the found maximum value. It is
 // initialized to 0 and is resulting integer type.
-//
-// The second output buffer contains the maximum value found. It is initialized
-// to the minimum representable value of the input element type. After being
-// populated by indexed_generic, this buffer is disgarded as only the index is
-// requested.
-//
-// The indexed_generic op updates both the maximum value and index if the
-// current value exceeds the running max.
-class ConvertAtenArgmaxOp : public OpConversionPattern<AtenArgmaxOp> {
+static std::pair<Value, Value>
+createArgMaxKernel(OpBuilder &b, Location loc, Value input,
+                   Optional<int64_t> dim, bool keepDim,
+                   RankedTensorType idxResType, RankedTensorType valResType) {
```

Comment:
nit: I think its better to flip the order of these two result types so that they match the order of the types of the output

Reply:
I know, I just meant that because this kernel calculation is made for `maxDimOp`, it should use the same order of arguments that `maxDimOp` has

Reply:
Done